### ISSUE TITLE
Camera traffic, when app in background. 

### ISSUE THREAD

**Issue by nick2525** [home-assistant/android#979]
**Home Assistant Android version:**
2.4.1-full

**Android version:**
10
**Phone model:**
Xiaomi mi 9
**Home Assistant version:**
0.115.3
**Last working Home Assistant release (if known):**
no
**Description of problem:**

When app in background, video streams from cameras does not stop stream. Almost all apps stops video stream at onPause() Android lifecycle, but HA does not. Now the problem is even more serious, because from the activities of the sensors, the system almost never kills the application. I have a lot of cameras and I wasted 70 gigabytes in a month because of this bug. Plus the battery runs out quickly.
**Traceback (if applicable):**

**Screenshot of problem:**
![Screenshot_2020-09-25-18-25-18-432_com miui securitycenter](https://user-images.githubusercontent.com/569343/94288159-c75da300-ff5f-11ea-87c5-c39e0c9ed4b8.jpg)

**Additional information:**
Solution: You should stop stream, when device goes onStop(), maybe communicate with HA server, and show video placeholder.

**Comment by dshokouhi** [comment]
You need to check that "automatically close connections" in your HA user profile is enabled.  The frontend actually handles this part as we are just a webview so they look to see if you have gone idle.

 You should also check that you are not going to the home screen when you have the more info panel open, another offender in this case is if you use the live view card on lovelace which plays video on the homescreen directly.

**Comment by nick2525** [comment]
1) I checked "automatically close connections"  - "Should we close the connection to the server after being hidden for 5 minutes?" - was on, but it seems 5 minutes too much for mobile. 
2) I use card on lovelace which plays video on the homescreen directly - as it is nice feature. 
3) Yes, HA is Webview, but HA is not website, here you can control server or frontend, and forward iOS /Android lifecycle to frontend, maybe even to backend, to lower the number of frames in the video, or use still image when APP is onStop() or onPause()

**Comment by barrymossel** [comment]
This started for me too after (one of the) latest update(s). Just spend 2,5 GB in 10 minutes. Never experienced this before.

**Comment by JBassett** [comment]
@nick2525  and @barrymossel  can you please include your yaml camera config and frontend config so we can attempt to recreate.  I have tried with my personal setup and I am unable to get the app to stream in the background.

**Comment by nick2525** [comment]
camera:
  - platform: mjpeg
    name: children_camera
    username: *****
    password: *****
    mjpeg_url: http://***.***.***.***/live/0/mjpeg.jpg
    
camera 3:
  - platform: mjpeg
    name: kitchen_camera
    mjpeg_url: http://***.***.***.***:56000/mjpeg    
        
camera 4:
  - platform: mjpeg
    name: slipping_camera
    username: admin
    password: admin
    mjpeg_url: http://***.***.***.***/live/0/mjpeg.jpg

camera 5:
  - platform: generic
    name: slipping_room_camera
    framerate: 2
    still_image_url: http://***.***.***.***/webcapture.jpg?command=snap&channel=1
    
camera 6:
  - platform: generic
    name: child_room_camera
    framerate: 2
    still_image_url: http://***.***.***.***/webcapture.jpg?command=snap&channel=1

**Comment by nick2525** [comment]
{
                    "background": "radial-gradient(crimson, skyblue)",
                    "cards": [
                        {
                            "camera_view": "live",
                            "entity": "camera.children_camera",
                            "title": "\u041a\u043e\u0440\u0438\u0434\u043e\u0440",
                            "type": "picture-entity"
                        },
                        {
                            "camera_view": "live",
                            "entity": "camera.child_room_camera",
                            "title": "\u0441\u043f\u0430\u043b\u044c\u043d\u044f",
                            "type": "picture-entity"
                        },
                        {
                            "camera_view": "live",
                            "entity": "camera.kitchen_camera",
                            "title": "\u043a\u0443\u0445\u043d\u044f",
                            "type": "picture-entity"
                        },
                        {
                            "camera_view": "live",
                            "entity": "camera.slipping_camera",
                            "title": "\u043b\u0435\u0441\u0442\u043d\u0438\u0446\u0430",
                            "type": "picture-entity"
                        },
                        {
                            "camera_view": "live",
                            "entity": "camera.slipping_room_camera",
                            "title": "\u0441\u043f\u0430\u043b\u044c\u043d\u044f",
                            "type": "picture-entity"
                        }
                    ],
                    "id": "cameras",
                    "theme": "dark-mode",
                    "title": "\u043a\u0430\u043c\u0435\u0440\u044b"
                }

**Comment by barrymossel** [comment]
```
camera:
  # - platform: foscam
    # ip: !secret ip_camera_kleren
    # name: Klerenkamer
    # username: !secret camera_kleren_username
    # password: !secret camera_kleren_password
  # - platform: ffmpeg
    # name: Lise
    # input: !secret url_camera_lise
  - platform: mjpeg
    name: Sophie
    mjpeg_url: http://192.168.1.43:8081/
  - platform: mjpeg
    name: Lise
    mjpeg_url: http://192.168.1.43:8082/
  - platform: mjpeg
    name: Hal
    mjpeg_url: http://192.168.1.43:8083/
```
The ones that were commented out are my old camera's which didn't have this problem. When I "updated" those the traffic started getting out of hand. Did some more investigation and I use MotionEye as when I load the rtsp feeds directly into HA I get a lag ranging from 4 to 60 sec. When I load the rtsp feed into MotionEye and let that process and stream it, there is virtually no lag (max 1 sec). But apparently the feeds use way too much bandwidth. When I load the stream url into a browser it uses lot's of data as well. So that's one of the issues for me. But no I turned the quality of the feeds to 25% and it still uses 100's of MB's per couple of minutes. Way more than the original feed passing through the Ewelink app (camera's are Sonoffs). 
But still, I believe the old camera's didn't do any of this anyway. Never noticed something weird on the data usage side of the HA app... So, could be two things I guess: very small feeds with low data usage, or no streaming in background.

And btw, if I force the app to close it doesn't stream and use data. If I just go back to Android home screen it does use bandwidth.

**Comment by barrymossel** [comment]
Lovelace:

```
          - type: picture-glance
            title: Sophie
            camera_image: camera.sophie
            camera_view: live
            aspect_ratio: 16:9
            entities:
              - binary_sensor.deursensor_sophie_contact      
              - light.sophie       
          - type: picture-glance
            title: Lise  
            camera_image: camera.lise
            camera_view: live
            aspect_ratio: 16:9
            entities: 
              - binary_sensor.deursensor_lise_contact
              - light.lise
```

**Comment by bram2202** [comment]
I got the same issue (#1017)
I also happen to have a Xiaomi mi 9(T pro)

HA config:
```
- platform: generic
  name: Hallway
  still_image_url: !secret camera_hallway_snap_url
  verify_ssl: false

- platform: generic
  name: Street
  still_image_url: !secret camera_street_snap_url
  verify_ssl: false

- platform: generic
  name: Livingroom
  still_image_url: !secret camera_livingroom_snap_url
  verify_ssl: false
```
Lovelace:
```
    cards:
      - aspect_ratio: 0%
        camera_image: camera.hallway
        camera_view: live
        entity: camera.hallway
        show_name: false
        show_state: false
        type: picture-entity
      - aspect_ratio: 0%
        camera_image: camera.livingroom
        camera_view: live
        entity: camera.livingroom
        show_name: false
        show_state: false
        type: picture-entity
      - aspect_ratio: 0%
        camera_image: camera.street
        camera_view: live
        entity: camera.street
        show_name: false
        show_state: false
        type: picture-entity
```

**Comment by nick2525** [comment]
My wife has Xiaomi mi 9T, the bug is also reproduced.

**Comment by ericlathrop** [comment]
I just got burned by this, wasted almost 1GB of my data plan. Using the Zoneminder integration, and Picture Glance Cards.

```
zoneminder:
  - host: zoneminder.mydomain.com
    ssl: true
    username: x
    password: x

sensor:
  - platform: zoneminder
    include_archived: false

camera:
  - platform: zoneminder
```

**Comment by bramkragten** [comment]
Fix: https://github.com/home-assistant/frontend/pull/7362

**Comment by nick2525** [comment]
Was it fixed in 0.117.0b1? Because the bug is still being reproduced(

**Comment by dshokouhi** [comment]
@nick2525 can you verify the frontend version you are on please?

**Comment by nick2525** [comment]
Frontend version: 20201021.1 - latest

**Comment by nick2525** [comment]
![Screenshot_2020-10-22-20-13-52-062_com miui securitycenter](https://user-images.githubusercontent.com/569343/96906915-281ac580-14a3-11eb-913f-c82e0cba1be9.jpg)

**Comment by dshokouhi** [comment]
@nick2525 thanks, and you waited 5 minutes for the connection to close right?

**Comment by dshokouhi** [comment]
BTW it looks like the linked issue was actually not the fix for this.  That was the reasoning for the bug closing and being re-opened.

**Comment by nick2525** [comment]
Some cameras disconnected after 5 minutes, but live view is not restored on those cameras. But if I pres on camera, stream works. But live view is not restored on app onResume()

**Comment by nick2525** [comment]
![Uploading Screenshot_2020-10-22-20-22-28-033_com.miui.securitycenter.jpg…]()

**Comment by dshokouhi** [comment]
live view being restored is unrelated to this issue....this issue is purely about making sure the data flow stops when the app is in the background after 5 minutes.

**Comment by nick2525** [comment]
![Screenshot_2020-10-22-20-44-29-006_com miui securitycenter](https://user-images.githubusercontent.com/569343/96910018-821d8a00-14a7-11eb-98da-5664d813779d.jpg)
It seems, that some camera didn't stop after ten minutes, because It too much data for sensors, but it 8 times less traffic when after one minute in background

**Comment by dshokouhi** [comment]
what are the results if you disable all your picture entity cards? does it 0 out then after 5 minutes?

**Comment by nick2525** [comment]
![Screenshot_2020-10-22-21-06-34-703_com miui securitycenter](https://user-images.githubusercontent.com/569343/96912174-97e07e80-14aa-11eb-8797-567e8dd868b3.jpg)

**Comment by nick2525** [comment]
It almost no when I disabled camera view live

**Comment by nick2525** [comment]
@dshokouhi @bramkragten @JBassett https://github.com/home-assistant/frontend/pull/7362 fixed all cameras, except one which has no stream, and only still_image_url, so 2 frame per second is about 240 KB/s, I disabled camera view live for each cameras, and only camera with only still_image_url and no stream produced traffic after 8 minutes  ```
"platform: generic
name: child_room_camera
framerate: 2
still_image_url: http://.../webcapture.jpg?command=snap&channel=1"
```

**Comment by nick2525** [comment]
It still reproduced on still_image_url only cameras, 
version | 0.117.2
frontend  20201021.4

**Comment by bendikrb** [comment]
I can confirm this issue, which exists on both mine and my partners phones - since 0.117 ish up until now (0.118.0b2).
The weird thing is that we're seeing data being counted as "foreground", although the app certainly hasn't been open/in use for the duration it would take to transfer this much data.

Most notably today, when it ran away using 1.9GB from 00:00 till about 07:30;
![Screenshot_20201115-131712_Settings_200](https://user-images.githubusercontent.com/152300/99184874-7770b080-2746-11eb-81ff-b2b6246062a9.jpg)

**Comment by nick2525** [comment]
@bendikrb Do you have still_image_url camera? Or you have steam?

**Comment by bendikrb** [comment]
> Do you have still_image_url camera? Or you have steam?

One of them is defined using the mjpeg platform, with both `still_image_url` and `mjpeg_url` defined;
```yaml
- platform: mjpeg
  name: front_door_lowres
  username: !secret vivotek_username
  password: !secret vivotek_password
  verify_ssl: false
  still_image_url: !secret vivotek_still_image_url
  mjpeg_url: !secret vivotek_mjpeg_lowres_url
```
The other one is set up using the onvif integration (using config_flow), and is proxied (this is the entity I'm using in lovelace);
```yaml
- platform: proxy
  entity_id: camera.backyard_onvif_hires
  name: backyard_hires
  max_stream_height: 1280
  cache_images: true
```

OT: The reason for proxying is because if I don't, and display the entity directly in lovelace, the ffmpeg parameters i have defined in the onvif settings isn't applied (dewarping of the 180° fisheye lens).

**Comment by dshokouhi** [comment]
If you are still experiencing this issue after you update to at least BETA 522 please also test once in Chrome and let us know if the issue occurs there as well.

**Comment by nick2525** [comment]
It seams it wasn't fixed on Xiaomi mi 9, I installed beta-522-d99daec-full , and ha consume 1.12 mb/s in Xiaomi security app. I don't have spit screen, and Xiaomi security app is resumed, it not transparent. I also tried to open Google translate App after HA, and when Xiaomi security app. Bug 🐛 is reproduced

**Comment by dshokouhi** [comment]
@nick2525 can you check in chrome and see if you can reproduce there please.

**Comment by nick2525** [comment]
![Screenshot_2020-11-17-01-55-20-018_com miui securitycenter](https://user-images.githubusercontent.com/569343/99318105-282a9d00-2878-11eb-88bf-89c56991a607.jpg)
Same problem on chrome

**Comment by nick2525** [comment]
![Screenshot_2020-11-17-02-05-21-016_amirz rootless nexuslauncher](https://user-images.githubusercontent.com/569343/99318795-7b511f80-2879-11eb-9aba-a1fabba8eb52.jpg) traffic from HA when I am at launcher

**Comment by bramkragten** [comment]
How are you exactly doing these tests? Can you make a step by step guide with timing? And what is the config of the problematic dashboard?

**Comment by dshokouhi** [comment]
@nick2525 also please ensure you are waiting 5 minutes for the connection to close as well, please be precise with your steps when you comment back.

**Comment by nick2525** [comment]
https://drive.google.com/file/d/1fD4z951965KTrxWSIvsIXk4jyztIzPFM/view?usp=drivesdk I uploaded video. Now I didn't wait 5 minutes, as though that onStop fix should work immediately. I have already posted my cameras config at beginning of bug

**Comment by dshokouhi** [comment]
@nick2525 this is in `onPause` please wait 5 minutes as thats intended behavior.  The feeds should stop after the connection closes in 5 minutes, not immediately.

**Comment by nick2525** [comment]
![Screenshot_2020-11-17-03-04-16-242_amirz rootless nexuslauncher](https://user-images.githubusercontent.com/569343/99323152-a5f3a600-2882-11eb-8f2c-b7af1b7c10b9.jpg) begin 
![Screenshot_2020-11-17-03-09-34-178_amirz rootless nexuslauncher](https://user-images.githubusercontent.com/569343/99323181-b6a41c00-2882-11eb-9276-934a6a2e580d.jpg) after 5 minutes some cameras stopped stream except still image url cameras. 
![Screenshot_2020-11-17-03-09-45-403_com android systemui](https://user-images.githubusercontent.com/569343/99323261-e4896080-2882-11eb-8451-72fec61bcbc0.jpg) nothing changed after last fix home-assistant/frontend#7362

**Comment by dshokouhi** [comment]
@nick2525 and just to confirm you see the exact same behavior in chrome with the 5 minute wait?

**Comment by nick2525** [comment]
Yes, same at Chrome 1.1 mb/s during 5 minutes, 200 kb/s after 5 minutes (some cameras were disconnected after 5 minutes but not all)

**Comment by bramkragten** [comment]
Ok, do you know what camera was and what camera wasn't stopped?

**Comment by nick2525** [comment]
@bramkragten Yes, I removed all cameras and when added again one by one, and such camera wasn't stopped

platform: generic
name: slipping_room_camera
framerate: 2
still_image_url: http://.../webcapture.jpg?command=snap&channel=1

Mjpeg cameras were stopped, like this
platform: mjpeg
name: children_camera
username: *****
password: *****
mjpeg_url: http://.../live/0/mjpeg.jpg

**Comment by TRusselo** [comment]
this happened to my father’s phone today with the companion app.
9 GB used in one day.
Pixel 4XL
He never opens the app. its used for location tracking only.
“auto close connection” has been on for months with no issue until today.
I do have some live view cameras on a lovelace page, but no issue until today on 1 phone, not all.

**Comment by nick2525** [comment]
@TRusselo Do your have still image url cameras? I will try also to report bug to Chrome developers

**Comment by TRusselo** [comment]
I have live view cameras on a lovelace page,  page 7.
The cameras are live feeds using the BlueIris HACS integration.

Not chrome.  Android companion app... or is that built on chrome?

This once happened to my phone back on November, 1 random day where I did not open app that day.
Then yesterday to my fathers phone. he never uses the app.  Its basically just used for location tracking.
Odd how it just happens out of nowhere randomly until our carrier sends us an SMS that we have gone over a limit, or the phone starts burning a hole in our leg....  reboot. gone. fixed.  until next time....

**Comment by dshokouhi** [comment]
> What's the reason it's intended to stream for 5 minutes after closing? Everything more than 10 seconds seems needless.

The HA frontend has an auto close connection setting in the user profile that is set to 5 minutes.  This is the same behavior for browsers too.  The app just respects this.  This bug also impacts Chrome as noted in other comments too.

**Comment by barrymossel** [comment]
I get that, but also for browsers it seems quite long. But could it help to have a different setting in frontend for mobile apps/browsers? 
Last month I had almost 20GB of background data (luckily through wifi due to corona), which could be dramatically lower if the connection closes after 10 seconds. I would think that people open a mobile view of HA to quickly check something or turn something on, then close the view and don't open the mobile view for quite some time (at least 5 mins).

**Comment by dshokouhi** [comment]
> But could it help to have a different setting in frontend for mobile apps/browsers?

Thats unrelated to the android app but I am sure you can start a discussion about it in the frontend repo because that is where the update will come from.



> Last month I had almost 20GB of background data (luckily through wifi due to corona), which could be dramatically lower if the connection closes after 10 seconds.

You can just remove cameras from the home page and keep them on a tab that isn't normally visible and it will greatly reduce the camera stream issue.  The issue here is that the camera is the last thing being observed so its kept open.  By ensuring the video doesnt even start on the tab it removes the issue.  Also as noted in this thread the connection closing didn't solve the issue so we are not sure where it is coming from.

**Comment by barrymossel** [comment]
> > But could it help to have a different setting in frontend for mobile apps/browsers?
> 
> Thats unrelated to the android app but I am sure you can start a discussion about it in the frontend repo because that is where the update will come from.
> 
Done that: https://github.com/home-assistant/frontend/discussions/8831
> > Last month I had almost 20GB of background data (luckily through wifi due to corona), which could be dramatically lower if the connection closes after 10 seconds.
> 
> You can just remove cameras from the home page and keep them on a tab that isn't normally visible and it will greatly reduce the camera stream issue. The issue here is that the camera is the last thing being observed so its kept open. By ensuring the video doesnt even start on the tab it removes the issue. Also as noted in this thread the connection closing didn't solve the issue so we are not sure where it is coming from.

My cameras aren't on the homepage. Actually when I open a tab with cameras traffic goes up significantly, then switch to one without and close the app, traffic keeps on that level. Closing the app eliminates the traffic. 

Wouldn't this whole issue be frontend related? When a stream is out of view (either because the tab/view has changed to another, or the browser/app closed), it should not use traffic.

**Comment by barrymossel** [comment]
Ah, I am not alone: https://github.com/home-assistant/frontend/issues/8630 & https://github.com/home-assistant/frontend/issues/5437

**Comment by barrymossel** [comment]
> Also as noted in this thread the connection closing didn't solve the issue so we are not sure where it is coming from.

Btw, traffic does stop for me after XX (probably 5) minutes.

**Comment by barrymossel** [comment]
For MotionEye users:

Don't use the streaming url and camera entity, but embed url with iframe card. When the camera is out of view (either closing the app or changing tab/view) stops camera traffic.
Unfortunately not possible with picture_glance to show certain entity states, but I will think of something else for that...

**Comment by TRusselo** [comment]
still happening

**Comment by bramkragten** [comment]
This might (probably?) be fixed in 2021.9

**Comment by abmantis** [comment]
> This might (probably?) be fixed in 2021.9

Great! Any pointer to the fix commit?

**Comment by R5fan** [comment]
Im not sure yet if this is indeed fixed in 2021.9, Im only upgrading now, but I need an emergency workaround in case this isnt fixed or happens again in the future; today I blew through my monthly datacap and incurred a 50 euro data charge in an hour, and I hadnt even turned on my phone's screen. I need something to restrict companion's data usage. I disabled "background data" but that doesnt seem to help.  

I can presumably totally block it with an android firewall, but then I might as well just uninstall it. 

Im trying to make a node red flow that monitors the app_rx_gb sensor and correlates it with being on mobile data, but somewhat ironically, that sensor doesnt seem to update in the background, unless something else changes (like my location or alarm clock?) And even if it did, Im not sure what I can do to stop the data, other then send myself a notification. Which Ill probably read too late.

**Comment by dshokouhi** [comment]
All known workarounds are mentioned in this issue

**Comment by R5fan** [comment]
2019.9 doesnt fix it for me.  Checked HA on my phone this morning for 5 seconds, closed the app and phone and and 30 minutes later it has used 1.5GB and keeps going up.

When I said workaround, I mean some emergency / brute force "solution", something that works independently of the companion app. An android app or setting of some kind to limit the impact this issue can have or just deny it access to mobile data at all. In android settings I disabled mobile background data usage:
![image](https://user-images.githubusercontent.com/1439530/133203016-afe18031-69cc-45d9-ae35-f53391091e19.png)

but it seems to think HA doesnt run in the background? In the app it says it has background access:
![image](https://user-images.githubusercontent.com/1439530/133203083-2d3e3b1b-1156-4edf-b8aa-97509404750d.png)

How do I revoke that? 

I need something other than uninstalling HA companion (a ton of automations depend on its sensors), and right now all I have is turning off mobile data on my phone whenever I leave the house.

**Comment by TRusselo** [comment]
> 2019.9 doesnt fix it for me. Checked HA on my phone this morning for 5 seconds, closed the app and phone and and 30 minutes later it has used 1.5GB and keeps going up.
> 
> When I said workaround, I mean some emergency / brute force "solution", something that works independently of the companion app. An android app or setting of some kind to limit the impact this issue can have or just deny it access to mobile data at all. In android settings I disabled mobile background data usage:
> 
> but it seems to think HA doesnt run in the background? In the app it says it has background access:
> 
> How do I revoke that?
> 
> I need something other than uninstalling HA companion (a ton of automations depend on its sensors), and right now all I have is turning off mobile data on my phone whenever I leave the house.

Well... couple things....
Android Settings > apps > home assistant > Battery > Allow background Activity
Android Settings > apps > home assistant > Battery > Optimize battery usage
 **_BUT...  disabling these options will mess with the sensors you NEED working._**
 
Im in the same boat.  So i removed the cameras from home assistant.  I use blue iris to check them instead.

But here are a couple things that were helping me before i removed them.

#1  never close  HA app by pressing the HOME button on your phone.  The home button does not close ANY app. just pushes it to the background.  in Android there are 2 ways to close apps.  1. press BACK BACK BACK BACK BACK ect....   until you see the home screen. or 2. press RECENT APPS button and swipe HA away.

#2 Another work-around i was trying for a while, in HA put all cameras on a single page.  when checking the cameras go to that page, then when done, change to a different page before closing app.  **see above about closing apps.

**Comment by R5fan** [comment]
> 1. press BACK BACK BACK BACK BACK ect.... until you see the home screen. or 2. press RECENT APPS button and swipe the app away

This doesnt help. I force closed HA and it would still gobble up bandwidth "in the background".

Regarding Blue Iris, I did the same, relying mostly on BI web interface, and now I managed to actually integrate individual BI streams and its web GUI in to HA. I wrote a little piece on it here:
https://www.reddit.com/r/homeassistant/comments/po34pg/finally_low_latency_low_bandwidth_high_quality/

This also completely solves this particular issue, as blue iris will stop streaming when the camera frame has no focus, and/or after a configurable time. Completely independently of HA.

This isnt a good or universal solution (especially for those not using BI), but its a great workaround for me, or anyone in the same boat;  and even if this bug finally gets fixed, it will be some time before I trust HA camera's with my bandwidth.

**Comment by R5fan** [comment]
Using iframes for camera streams to work around this bug has at least problem I wasnt aware off; it stops android from entering sleep mode, which is particularly annoying on wall mounted tablets. 

I worked around that, and that solution may also work -or at least help- for this bug,  if you use regular picture entity cards for camera's: whenever I want the tablet to sleep, I redirect companion to a dedicated view that has no iframes (or camera's). I dont know for sure if this will stop regular camera entities from streaming in the background, but it may help and it certainly stops UI3 streams which only stream when they are in the foreground. 

This behavior has an unexpected benefit for wallpanels too; instead of having to keep sending notifications to keep my tablets awake while im in the room or the lights are on, the iframe does that for me now. So I can pretty much turn the screen on or off with one command. Full node red flow here:
https://www.reddit.com/r/homeassistant/comments/ppwln7/node_red_flow_to_allow_ha_tablets_to_enter_sleep/

**Comment by nick2525** [comment]
now the camera stream picture dies if HA is open for a very long time.  For example, I have a tablet hanging on the wall.  I would like the stream to automatically restart

**Comment by dshokouhi** [comment]
> now the camera stream picture dies if HA is open for a very long time. For example, I have a tablet hanging on the wall. I would like the stream to automatically restart

Does that happen in chrome for android?  I think there was some frontend websocket issues in HA core 2021.12 that were fixed in the latest point release.

**Comment by lsdlsd88** [comment]
this is still an issue. App on Android even in background pulling video stream all the time

**Comment by TRusselo** [comment]
yeah this did not happen to me in chrome.
HASS companion app

**Comment by zSprawl** [comment]
I’m also having the issue anytime a camera entity is on the dashboard, it will eventually crash the app. The more camera entities, the fast it manifests.

**Comment by dshokouhi** [comment]
> I’m also having the issue anytime a camera entity is on the dashboard, it will eventually crash the app. The more camera entities, the fast it manifests.

That is actually not the same issue as this one, as far as I know the app is not crashing here because there is background data occurring and if the app crashes after a short while I dont think anyone will see GB of data like above. Once an app crashes it is no longer connected to webview. If you are seeing crashes please open a new issue and fill out the template provided with the crash log. In your case check to see if there is a Recent Crash tab when you pull hte app logs for the template. If it is there that is the log we will need.

**Comment by radhoo2k10** [comment]
![Screenshot_20230321_182315_com huawei systemmanager](https://user-images.githubusercontent.com/78762281/226691532-d08bf158-005f-4756-a19f-534b1f20a3f9.jpg)
![Screenshot_20230321_182321_com huawei systemmanager](https://user-images.githubusercontent.com/78762281/226691548-4cf58c55-725d-486a-a4bb-d59a374382a6.jpg)
Still an issue as of 21th of March. Huawei android phone. Latest companion app and latest HA on server. 2 camera streams on page 1.

**Comment by Jeanne-D4RK** [comment]
Still an issue as of May 2024

![Screenshot_20240529_190840_Settings.jpg](https://github.com/home-assistant/android/assets/61759320/1cbc7930-094b-4201-a940-389edf67b71f)

**Comment by rosicenko** [comment]
Same issue happening to HA app running on my NSPanel Pro and absolutely trashing its RAM.

**Comment by kariudo** [comment]
It is insane that this is still an open issue after over 4 years... I just ran into this when I got a "you're out of data" notice because I briefly opened a tab I have on one of my dashboards with 10 cameras on it, then switched out of the app and put my phone away... when from the data usage recording on my phone I can see caused it to promptly use up 2GB while I was out.

**Comment by 5of9** [comment]
Absolte, should become a higher priority after 5 years. 6gb traffic in background, thats really a huge bug!

**Comment by kariudo** [comment]
> Absolte, should become a higher priority after 5 years. 6gb traffic in background, thats really a huge bug!

I put in a PR #5051 that solves the issue for me by unloading the webview when the app is paused (minimized or screen off), but theres apparently quesitonable interest in fixing this because the source of the problem isn't specificailly the app (even if the app can fix it). Might want to comment there if you're interested, they don't perceive this as a problem for the "end user" it seems. I've been running that branch's build the last few days (prevents a number of resource waste issues in my use).

**Comment by abmantis** [comment]
From my testing, https://github.com/home-assistant/frontend/pull/26235 should fix the issue for cameras in the background.

**Comment by dshokouhi** [comment]
Thank you for the taking the time to fix this from the frontend @abmantis !!  So the next HA core beta (2025.8) should allow for others to confirm with their setups.

**Comment by dshokouhi** [comment]
just wanted to check in to see if anyone is on the HA core 2025.8 beta and can comment on how the fix is working for them

**Comment by TantrumV4** [comment]
@dshokouhi I am already on the 2025.8.1 release and I also have the Live card for my camera. The problem seems to be still the same.

<img width="1008" height="2244" alt="Image" src="https://github.com/user-attachments/assets/02ad611d-7e8e-4692-be82-9893f9b75555" />

**Comment by abmantis** [comment]
@TantrumV4 which card are you using for the camera?

**Comment by MikeVensel** [comment]
https://github.com/home-assistant/android/pull/5051 added an option to unload the webview when the app was minimized but it seems to be in limbo because optional settings are bad or something?

**Comment by dshokouhi** [comment]
> https://github.com/home-assistant/android/pull/5051 added an option to unload the webview when the app was minimized but it seems to be in limbo because optional settings are bad or something?

Please see the above comments another fix that is more appropriate is out already.

**Comment by kariudo** [comment]
> > [#5051](https://github.com/home-assistant/android/pull/5051) added an option to unload the webview when the app was minimized but it seems to be in limbo because optional settings are bad or something?
> 
> Please see the above comments another fix that is more appropriate is out already.

https://github.com/home-assistant/android/issues/979#issuecomment-3109523506 may help with a specific case (though apparently not per @TantrumV4 reporting above; this @abmantis change seems to only attempt to stop and unload specifically the "built-in" webRTC component in the frontend, which not only won't help anyone not using that component, it won't help with all the other sources of resource waste when backgrounded (as likely exemplified in the above screenshot from @TantrumV4). It's possible the referenced change will result in an improvement for some cases here, but it does not resolve the full issue of background resource usage/waste.

For comparison to his screenshot, I have been running the change from #5051 since February and it has completely solved the problems for me, see my screenshot from the last day since a charge etc. (and yes I have background sensor data being submitted):

<img height="100" alt="Image" src="https://github.com/user-attachments/assets/7d9e0651-643a-4fd2-9f87-f0f2c0335af5" />

<img width="559" height="453" alt="Image" src="https://github.com/user-attachments/assets/3515c62a-7cd1-43a8-a3c5-ce759ec5b129" />

Until the frontend can actually stop every resource consumer when backgrounded (which is unlikely given the existence and need for custom components/HACS), then I still see a need for the optional ability to unload the WebView as in #5051. I don't understand the unwillingness to include an optional setting for those that clearly have expressed they want it.

Personally I think `<1%` is better than `50%` usage.

**Comment by abmantis** [comment]
> Personally I think `<1%` is better than `50%` usage.

That is true, but users should not have to enable a setting to fix this kind of issue, because then we are not actually fixing the issue. A better approach would be to identify what are the other resources that use too much background data, and fix it in the same way that was done for cameras.

Also, IIRC this issue would also happen when you use HA in Chrome on Android. So having the setting would be a workaround for the app but not for the browsers.

**Comment by MikeVensel** [comment]
Why can't both fixes co-exist? The frontend handles fixing some of the resource usage but clearly not even close to all of it and #5051 was already an optional setting and lets you further save resources if that is an issue for you.

**Comment by kariudo** [comment]
> That is true, but users should not have to enable a setting to fix this kind of issue, because then we are not actually fixing the issue. A better approach would be to identify what are the other resources that use too much background data, and fix it in the same way that was done for cameras.

If content in applications could be relied upon outside of a projects direct control to not consume extra resources, Google wouldn't have so many settings like "Allow running in background" and multiple levels of "battery usage" to force things to not consume finite resources of a device like a phone.

Sure it would be nice if "it just worked and the front end would never waste resources", but I think it is completely unrealistic to expect every component (especially HACS/custom/etc.) to support fully preventing the use of background data each, clearly this is already a problem, and already took 5 years for maybe one source of the wasted resources to be even partially addressed (I am not convinced this does given above reports).

I still so no reason to not include the optional setting for those that want more complete prevention of wasted background resources from the WebView. Saying "it's the frontend and other components problem to sort out each case" does not help us, the users. Especially when a solution exists and is offered for those that are asking for it.

The solution used is documented by Google specifically because a WebView can't be trusted to otherwise release all its resources. The proof is in the pudding here, @TantrumV4 has shown the recent offered solution does not fix the problem, and other users like @MikeVensel have said they want the more complete optional setting.

It feels like there is a strong lack of sympathy for those with limited data (in many places it is very expensive) and a desire to save their battery usage, when providing the option hurts no one. I understand and agree it would be nice if it wasn't a problem that needed a solution at this end... but it is.

**Comment by NikoGrub** [comment]
@kariudo could you please tell me how to enable the "Unload when the app is paused" setting?

My Pixel 7's battery life has been so affected by the constant draining and charging that I need a new smartphone now.

<img width="1079" height="1045" alt="Image" src="https://github.com/user-attachments/assets/17b65689-b8c0-4cbb-9c77-c68ea806a1ef" />

Edit: 
I recognized that this function was never released for public. That's sad.

**Comment by kariudo** [comment]
> [@kariudo](https://github.com/kariudo) could you please tell me how to enable the "Unload when the app is paused" setting?
> 
> My Pixel 7's battery life has been so affected by the constant draining and charging that I need a new smartphone now.
> 
> Edit: I recognized that this function was never released for public. That's sad.

Yeah, currently it only exists in the #5051 PR (which is currently ignored/closed) and my fork where I added that feature. It is not in any published version unless they decide to change their stance here and accept the optional feature. Sorry, I wish they would.

**Comment by sevmonster** [comment]
I don't manage cameras in my HA at all, let alone on my dashboard, so https://github.com/home-assistant/frontend/pull/26235 doesn't affect me—the app is still eating my battery quite aggressively. I've thought my battery health was bad for a long time now and that it was a normal for it to drain so quickly, but I guess that's not the case. If anything, the frequent draining of my battery may have well contributed to the worsening health of it overall. That's awesome!

One of the concerns raised by the team above was that implementing #5051 would not help users much, since they would need to do research on why their battery/data was getting gobbled up in the first place, and only after would they understand the source of the problem and discover the added toggle by e.g. ending up on this page. This is mega bad UX, and I understand why you all might want to avoid such an approach.

However, you must also consider the inverse: **Users should not have to jump through all these hoops to understand they have an issue**, and it is **extremely unhelpful** to not have any immediate solution for their issues **regardless of if they had to go looking for it first.** Yes, it's better the root cause be fixed over providing a workaround, but taking that position at exclusion of all others is egregious if not borderline malicious, when *the issue has existed for half a decade with only recent developments toward a potential solution.*

There are ultimately three things to consider with all this:
- Battery life and potentially data usage is spiking with the HA app due to background activity. This is a *code problem.*
- Users are being affected by this issue that they have no knowledge of beforehand, and must do research out-of-band to understand it. This is a *messaging problem.*
- There was a working interim solution posed to the issue, but it was rejected in favor of finding the root cause and fixing it instead. While noble, this is nonetheless a *management/vision problem,* as it *forsakes the users currently affected by the issue* until it is fixed, with no timeline on the fix.

To play devil's advocate, yes, there are legitimate, direct reasons to not implement #5051:
- The feature will be useless for users until they realize they need it.
- It's primarily intended to be a workaround for something that isn't working as expected, and having users rely on it reduces the pool of users to test fixes and provide feedback.
- It sets a precedent that may be unwanted, such that removing the toggle later after things are working correctly may spark ire.
- The help text on the toggle does not explain the nuance behind why its existence as a feature is justified, leading to it being enabled by users that either don't understand what it does or its side effects, or don't need it at all. This could lead to further problems that would also need to be fixed.
- A feature/codepath with nothing directing the user to it will end up unused and neglected by both users and developers. Dead code leads to a worse UX when that code eventually causes problems. Plus it's yet another thing that the lead devs must think about for potentially little to no benefit.

...But most of this hinges on *messaging.* Better messaging would allow the feature to be known about, would direct users to use it and prevent the codepath from being underutilized, and would inform users that the feature is likely temporary and shouldn't be relied upon forever, and would help absolve the developers if it doesn't work exactly right. The help text itself can be adjusted to better explain the reasoning behind the toggle, or additional text in other places can help justify it. Point is, if the *code problem* can't be solved in a reasonable amount of time, then it would behoove the *project managers* to act in such a way that prioritizes the immediate sanctity of the user; and if *messaging* is a considerable reason behind it such not being done, then **fix the messaging**.

This issue in particular has been open for over half a decade now and is still unsolved in a general sense: the app can still eat battery, and I assume data. Users do not know this (or at least I didn't) and are hit by this issue unawares. And it seems that, without a feature like #5051, I and others still have no solution (workaround or not) to the issue. My own solution might just to be to uninstall the app and add a shortcut to my HA instance from my homescreen, which is really not what I want to do, but having my battery go flat all the time is not something I can tolerate if I can help it. Furthermore, users may simply naïvely think that "yeah the HA app eats battery, that's just a thing that happens" when really, it's a bug.

Rather than just worry about fixes/workarounds and their viability, it is far more productive to consider the UX of the average user that might be experiencing issues they don't know about or understand, like I was. If #5051 were implemented, I could at the very least have a solution to my problem, even if it's not elegant, while the people invested in fixing the root cause can continue to do so, and interested testers can provide their feedback. And past that PR, better in-app messaging could help inform users to the whole situation.

---

As for this issue, the "suspend background connections" feature is already enabled in my user settings, yet the app is still absolutely devouring my battery.

<details>
  <summary>Screenshots</summary>
  <img width="1080" height="2092" alt="Image" src="https://github.com/user-attachments/assets/dfc0e5de-63c5-4681-9293-038aeeb4ad89" />
  <img width="1080" height="2092" alt="Image" src="https://github.com/user-attachments/assets/3b15a0df-46bc-4d6f-9597-45ffd4225426" />
  <img width="1080" height="2092" alt="Image" src="https://github.com/user-attachments/assets/33189953-1ea2-4099-8b47-bd9367dd76eb" />
  <img width="1080" height="2092" alt="Image" src="https://github.com/user-attachments/assets/1aae11af-bb37-45cc-8540-467bc6934f0c" />
</details>

Further points to consider:
- I *never* open the app, I only use the Device Controls native integration and Quick Access tile to manage my smart lights.
- I had no idea killing the frontend after a timeout was a feature, and I wouldn't expect such a feature to be in user settings of all places. I assume it's enabled by default or something, because I don't remember enabling it.
    - It being something that we as end users should ever need to consider is in itself strange to me. Is there a reason someone might want the frontend to continue in the background, especially in the mobile app?

**Comment by Cedric-Boucher** [comment]
It's 2026! Why is this still an issue!? I'm on the latest version of home assistant server (docker) and companion app (android up to date on pixel 9 pro XL and HA companion version 2025.11.4-full), I DON'T have ANY cameras, I don't have any pictures or any kind of media on any of my lovelace dashboards. I just have a few simple dashboards with some numbers and plots. Home Assistant companion app has used nearly 1GB of my data in the past 24 hours, and I haven't opened it while on data at all! I only have 1GB of data available per month and the next app with highest usage per month only uses a few MB normally. This is absolutely UNREASONABLE!!! This NEEDS to be fixed with high priority, it is NOT ok to be burning through unknowing users' data and then delay solving this issue for 6 years!!!

**Comment by royalhorror** [comment]
Why is this issue beeing ignored? I have to force stop the app to stop the sound of my camera stream from playing. I need the app to run in the background for widgets and notifications to work.

**Comment by sevmonster** [comment]
Your guess is as good as mine. I was forced to uninstall the app, personally. It is unusable in its current state, unless you remember to kill it regularly (I don't). I just made a shortcut on my home screen to open the web UI.

**Comment by MikeVensel** [comment]
@royalhorror @sevmonster https://github.com/home-assistant/android/pull/5051 proposed a solution to this issue but was rejected for what seemed like silly reasoning. If you are so inclined, you could probably build the APK from the fork and load it onto your devices.

**Comment by TimoPtr** [comment]
> Why is this issue beeing ignored? I have to force stop the app to stop the sound of my camera stream from playing. I need the app to run in the background for widgets and notifications to work.

It's not ignored a solution has been pushed some time ago on the frontend you probably want to open an issue there

**Comment by sevmonster** [comment]
@MikeVensel I'm aware, see [my post from half a year ago](https://github.com/home-assistant/android/issues/979#issuecomment-3477769998).

And yes, while that might be a solution, I'm not bothered enough to set up an Android build environment just for this, when I can use the web UI. All I really do is toggle lights. Would be nice to have native integration, but eh.

This issue certainly is keeping me from investing more in Home Assistant, and I'd be open to alternative platforms.

@TimoPtr That isn't at all helpful to the people here. What solution? Did it affect the app? If so, why wasn't this (very long-running and still active) issue not closed?

**Comment by abmantis** [comment]
> [@TimoPtr](https://github.com/TimoPtr) That isn't at all helpful to the people here. What solution? Did it affect the app? If so, why wasn't this (very long-running and still active) issue not closed?

Because there are multiple causes for this issue. One of the common ones (camera streams) was solved as @TimoPtr pointed out, but there appear be other causes (even with cameras, if you use custom camera cards).

If anyone finds the root cause for any of the background traffic issues and fixes it, it would be great.

**Comment by royalhorror** [comment]
> > [@TimoPtr](https://github.com/TimoPtr) That isn't at all helpful to the people here. What solution? Did it affect the app? If so, why wasn't this (very long-running and still active) issue not closed?
> 
> Because there are multiple causes for this issue. One of the common ones (camera streams) was solved as [@TimoPtr](https://github.com/TimoPtr) pointed out, but there appear be other causes (even with cameras, if you use custom camera cards).
> 
> If anyone finds the root cause for any of the background traffic issues and fixes it, it would be great.

I see no references to the fix in his post, just a statement that it is fixed.

I just now had to force close the HA app (2026.6.5) to stop the Frigate camera stream from running in the background. So if the camera stream issue is solved, it's not solved for the new auto-generated Overview dash. I opened the stream through "Security" in "Overview".

**Comment by sevmonster** [comment]
> > > [@TimoPtr](https://github.com/TimoPtr) That isn't at all helpful to the people here. What solution? Did it affect the app? If so, why wasn't this (very long-running and still active) issue not closed?
> >
> > Because there are multiple causes for this issue. One of the common ones (camera streams) was solved as [@TimoPtr](https://github.com/TimoPtr) pointed out, but there appear be other causes (even with cameras, if you use custom camera cards).
>
> I see no references to the fix in his post, just a statement that it is fixed.

There have been multiple fixes pushed related to data use by HA in general, which does affect the app. The main interaction between the app and the user is an Android WebView that simply loads the frontend, so fixes to the frontend propagate down. Look through the comments above and you'll find some.

The main focus for the team has been preventing camera/audio/data streams from running when the web component isn't in focus. As far as I've read, this *has* been largely fixed. And as stated, that *does* affect the app. But it hasn't been fixed entirely if the posts here are anything to go by. I assume there's still configuration states in the built-in or third-party cards etc. that may still run in the WebView when backgrounded, which is why there's still people here having data issues.

As for me and some others, the data use is not the main problem, it's that the WebView stays loaded and apparently busy in general, which I assume prevents the phone from going into low power mode (doze etc). Some phones aggressively kill background processes like this, which is why this may not be more widespread, but my phone does not.

> If anyone finds the root cause for any of the background traffic issues and fixes it, it would be great.

Both the battery and data issues stem from the same source: the WebView stays alive and active when the app is no longer in the foreground. A fix for this was provided but was not accepted, I went over that in my earlier comment.

I believe this is *not* a frontend issue and does not need to be fixed there. And if custom cards etc. can simply cause it to happen again, no amount of fixing the frontend will help. Ultimately we need a solution in the app for people that have setups susceptible. And we already have a mostly usable PR for that.

**Comment by TimoPtr** [comment]
> > > > [@TimoPtr](https://github.com/TimoPtr) That isn't at all helpful to the people here. What solution? Did it affect the app? If so, why wasn't this (very long-running and still active) issue not closed?
> > > 
> > > 
> > > Because there are multiple causes for this issue. One of the common ones (camera streams) was solved as [@TimoPtr](https://github.com/TimoPtr) pointed out, but there appear be other causes (even with cameras, if you use custom camera cards).
> > 
> > 
> > I see no references to the fix in his post, just a statement that it is fixed.
> 
> There have been multiple fixes pushed related to data use by HA in general, which does affect the app. The main interaction between the app and the user is an Android WebView that simply loads the frontend, so fixes to the frontend propagate down. Look through the comments above and you'll find some.
> 
> The main focus for the team has been preventing camera/audio/data streams from running when the web component isn't in focus. As far as I've read, this _has_ been largely fixed. And as stated, that _does_ affect the app. But it hasn't been fixed entirely if the posts here are anything to go by. I assume there's still configuration states in the built-in or third-party cards etc. that may still run in the WebView when backgrounded, which is why there's still people here having data issues.
> 
> As for me and some others, the data use is not the main problem, it's that the WebView stays loaded and apparently busy in general, which I assume prevents the phone from going into low power mode (doze etc). Some phones aggressively kill background processes like this, which is why this may not be more widespread, but my phone does not.
> 
> > If anyone finds the root cause for any of the background traffic issues and fixes it, it would be great.
> 
> Both the battery and data issues stem from the same source: the WebView stays alive and active when the app is no longer in the foreground. A fix for this was provided but was not accepted, I went over that in my earlier comment.
> 
> I believe this is _not_ a frontend issue and does not need to be fixed there. And if custom cards etc. can simply cause it to happen again, no amount of fixing the frontend will help. Ultimately we need a solution in the app for people that have setups susceptible. And we already have a mostly usable PR for that.

Thanks for the summary. We rejected the "fix" because it was not a fix it was a patch that would hide issues that needs to be addressed. Like you said something is keeping the WebView alive when in background, I have multiple servers and I don't see any issue at the moment even with most of the sensor enabled. I suspect like you mentioned a third party misbehaving, I don't have a good way to track down which one except disabling then one by one and see if it improves and then reports the issue to the third party (or frontend if it's in the official codebase). It would be awesome if we could detect wrong behavior in the frontend but I have no idea if it's possible TBH.

**Comment by sevmonster** [comment]
I have a vanilla installation without custom cards. I only have smart lights added. The app would regularly drain my entire battery over the course of a few hours. Of course, I haven't tried since the frontend fixes, so I don't know if the issue is resolved for me.

**Comment by sevmonster** [comment]
Well, I reinstalled the app:

<img width="944" height="1475" alt="Image" src="https://github.com/user-attachments/assets/42d88ea2-d88c-4fd7-8bce-fd79d578bd67" />

And this is what it's done over the last two hours. While 100% in the background. So I'll be disabling it again.

App version 2026.6.5-minimal; HA containerized, version 2026.6.4, frontend version 20260429.4.

This has been going on for years in some form or another. I, and everyone else here, would rather take a "patch that would hide issues that needs to be addressed" over this dumpster fire. No one is happy with the current situation, not the users or developers, so why not at least make one party happy with a temporary patch?

**Comment by MikeVensel** [comment]
This idea of addressing it on the front end side is a little ridiculous in my humble opinion. It seems improbable that every component both first and third party will be made to work efficiently enough for this app to work for people.

I really don't understand how the idea of unloading the web view in a minimized app is in any way a hack or a workaround. The app is in the background there should be no running web view. However, I'm tired of engaging with this issue when the response is always the same.

**Comment by TimoPtr** [comment]
I'm looking for people to try this approach https://github.com/home-assistant/android/pull/7135 by installing the debug APK available in `Checks -> Pull Request` scroll to the end and find the APKs and install the debug full one.

I'm trying to pause the WebView when the app is stopped.

**Comment by sevmonster** [comment]
So far, it's using so little battery that it doesn't even show up on the graphs. That's with *Always* mode enabled for connectivity to the HA server (as I do not have Google services and use the `minimal` release).

I hope it also helps those with high data use as well.

**Comment by andonevris** [comment]
In my experience the issue occurs only after viewing any page with camera feeds that have audio enabled. 

I have set up an alert  to notify me when audio is active. There is a sensor on the app you can activate called "music playing" which alerts when webview is active. 

My findings are

Open any page on the app WITHOUT a camera feed. 
- Music playing sensor turns on. 
- If you close the app yourself the music on sensor turns off immediately. 
- If you background the app the music on sensor will stay on for a while then turn off after a few minutes (I think this might be the new behaviour added to combat this issue?)

Open any page on the app WITH a camera feed. 
- Music sensor turns on
- If you close the app manually sensor stays on permanently 
- If you background the app the sensor stays on permanently
- Only killing the app will actually turn off the sensor

I use the sensor now with a timer to alert me if music has been playing too long so I can kill the app before it kills my battery.

**Comment by TimoPtr** [comment]
> So far, it's using so little battery that it doesn't even show up on the graphs. That's with _Always_ mode enabled for connectivity to the HA server (as I do not have Google services and use the `minimal` release).
> 
> I hope it also helps those with high data use as well.

You are connected to the same servers right? Open dashboards with camera and so on?

**Comment by sevmonster** [comment]
> You are connected to the same servers right? Open dashboards with camera and so on?

I have *never had any cameras added to HA*. I have only ever had smart lights. Both my old lovelace dashboard and the new overview caused the same massive battery drain. Like I said before, the battery drain and excessive data use were seemingly linked to the same problem: the WebView not unloading when no longer in use. Doesn't really matter the content of the view in my experience for it to be an issue in some way. Data only becomes an issue if you have cards that use lots of data.

And yes of course, same server and versions as my last test, unchanged dashboard.

**Comment by TimoPtr** [comment]
> > You are connected to the same servers right? Open dashboards with camera and so on?
> 
> I have _never had any cameras added to HA_. I have only ever had smart lights. Both my old lovelace dashboard and the new overview caused the same massive battery drain. Like I said before, the battery drain and excessive data use were seemingly linked to the same problem: the WebView not unloading when no longer in use. Doesn't really matter the content of the view in my experience for it to be an issue in some way. Data only becomes an issue if you have cards that use lots of data.
> 
> And yes of course, same server and versions as my last test, unchanged dashboard.

From your previous message I see that you are probably not running a stock ROM or at least not running one with the Play services. It could be that on this ROM the WebView implementation you have as an issue. I do tests on many phones daily since at least 2y and I don't see any battery drain issue TBH, but all my devices have Google services and the WebView is from the Play store.

On my main device using HA with 3 cameras on my default dashboard and the advance card, I only have 2GB of usage since March. In the last 24h HA used 10% of batterie with a widget updating every seconds on my home screen.

**Comment by sevmonster** [comment]
Many, if not most Android phones have very forceful anti-background features to save battery. OnePlus, Oppo, Xioami, really any Chinese ROM will kill any background WebViews, and full activities after a short while. Samsung is up there too with the best of them. This is all [very widely publicized](https://dontkillmyapp.com/), so the fact that Android (read: AOSP) doesn't, by default, aggressively kill backgrounded activities, is more of how Android *should* work, and not something foreign. Pixels also lack these aggressive features, and you can find people complaining about Pixel battery life likely as a result of that. Probably also why a lot of battery-related stuff got merged in the last few Android major releases, so Google can mainline it for their Pixels.

But I digress. I use [Graphene](https://grapheneos.org/), which does use a Chromium fork [Vanadium](https://github.com/GrapheneOS/Vanadium), so it is possible it could be doing something wrong. But in any case, my issue appears to be solved on the PR build.

I would be curious what the other users here have phone-wise to correlate your claim.

**Comment by abmantis** [comment]
> So far, it's using so little battery that it doesn't even show up on the graphs. That's with _Always_ mode enabled for connectivity to the HA server (as I do not have Google services and use the `minimal` release).
> 
> I hope it also helps those with high data use as well.

Have you tried setting the persistent connection to "Never" for a while? Does it still use a lot of background data?

**Comment by royalhorror** [comment]
> In my experience the issue occurs only after viewing any page with camera feeds that have audio enabled.
> 
> I have set up an alert to notify me when audio is active. There is a sensor on the app you can activate called "music playing" which alerts when webview is active.
> 
> My findings are
> 
> Open any page on the app WITHOUT a camera feed.
> 
> * Music playing sensor turns on.
> * If you close the app yourself the music on sensor turns off immediately.
> * If you background the app the music on sensor will stay on for a while then turn off after a few minutes (I think this might be the new behaviour added to combat this issue?)
> 
> Open any page on the app WITH a camera feed.
> 
> * Music sensor turns on
> * If you close the app manually sensor stays on permanently
> * If you background the app the sensor stays on permanently
> * Only killing the app will actually turn off the sensor
> 
> I use the sensor now with a timer to alert me if music has been playing too long so I can kill the app before it kills my battery.

This might be the same issue as I have experienced. The audio from the camera feed kept playing untill i force killed the app.



> Many, if not most Android phones have very forceful anti-background features to save battery. OnePlus, Oppo, Xioami, really any Chinese ROM will kill any background WebViews, and full activities after a short while. Samsung is up there too with the best of them. This is all [very widely publicized](https://dontkillmyapp.com/), so the fact that Android (read: AOSP) doesn't, by default, aggressively kill backgrounded activities, is more of how Android _should_ work, and not something foreign. Pixels also lack these aggressive features, and you can find people complaining about Pixel battery life likely as a result of that. Probably also why a lot of battery-related stuff got merged in the last few Android major releases, so Google can mainline it for their Pixels.
> 
> But I digress. I use [Graphene](https://grapheneos.org/), which does use a Chromium fork [Vanadium](https://github.com/GrapheneOS/Vanadium), so it is possible it could be doing something wrong. But in any case, my issue appears to be solved on the PR build.
> 
> I would be curious what the other users here have phone-wise to correlate your claim.

I'm also on GrapheneOS with no play services and the minimal release of HA.

**Comment by sevmonster** [comment]
> > I hope it also helps those with high data use as well.
> 
> Have you tried setting the persistent connection to "Never" for a while? Does it still use a lot of background data?

I meant, my problem is battery related, not data, and I hope those with data problems are also helped.

**Comment by abmantis** [comment]
> > > I hope it also helps those with high data use as well.
> > 
> > 
> > Have you tried setting the persistent connection to "Never" for a while? Does it still use a lot of background data?
> 
> I meant, my problem is battery related, not data, and I hope those with data problems are also helped.

Disabling the persistent connection should also improve battery. Have you tried it for a day or so?

**Comment by sevmonster** [comment]
> Disabling the persistent connection should also improve battery. Have you tried it for a day or so?

Having it enabled, as it was before on the mainline release, while [having battery use so low that it doesn't even register on the chart](https://github.com/home-assistant/android/issues/979#issuecomment-4900704052), should tell you that the problem had nothing to do with that setting.

<img width="1080" height="199" alt="Image" src="https://github.com/user-attachments/assets/1a7e3678-57e7-45fb-870f-0767429c57b7" />

I have in fact already tested if HA connectivity was the cause, last November when I first found this issue. I disabled everything I could, and the problem still occured.

**Comment by TimoPtr** [comment]
> > Disabling the persistent connection should also improve battery. Have you tried it for a day or so?
> 
> Having it enabled, as it was before on the mainline release, while [having battery use so low that it doesn't even register on the chart](https://github.com/home-assistant/android/issues/979#issuecomment-4900704052), should tell you that the problem had nothing to do with that setting.
> <img alt="Image" width="1080" height="199" src="https://private-user-images.githubusercontent.com/16788697/618838347-1a7e3678-57e7-45fb-870f-0767429c57b7.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODM1MjEzOTQsIm5iZiI6MTc4MzUyMTA5NCwicGF0aCI6Ii8xNjc4ODY5Ny82MTg4MzgzNDctMWE3ZTM2NzgtNTdlNy00NWZiLTg3MGYtMDc2NzQyOWM1N2I3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA3MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwNzA4VDE0MzEzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTllMjcyNGI2YzQ2NWRjMzYyYjZjNzRjYjU1NDA3YjAyNDEzMzA2OWJjNjg5MzY3Nzg2ZGZhZmU5YmMwMjc4YjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9._j2l1e6pgPUfp7aBEaDeRZbJwgde9q-o407jBWXYTzc">
> 
> I have in fact already tested if HA connectivity was the cause, last November when I first found this issue. I disabled everything I could, and the problem still occured.

So the PR did solve your issue in the end?

**Comment by sevmonster** [comment]
> So the PR did solve your issue in the end?

Yes. My battery is no longer drained. As I said, I hope it would help those here with data issues too.

**Comment by TimoPtr** [comment]
The fix is in the 2026.7.2 that is currently under review on the Play Store. Once it's released I would like to close this issue.
